data-prepper-core/src/main/java/com/amazon/dataprepper/parser/config/MetricsConfig.java

```diff
@@ -60,21 +81,78 @@ public JvmThreadMetrics jvmThreadMetrics() {
         return new JvmThreadMetrics();
     }
 
+    private void configureMetricRegistry(final MeterRegistry meterRegistry) {
+        meterRegistry.config()
+                .commonTags(Collections.singletonList(
+                        Tag.of(SERVICE_NAME, getServiceNameForMetrics())
+                ));
+
+    }
+
+    @Bean
+    public Optional<MeterRegistry> prometheusMeterRegistry(
+            final DataPrepperConfiguration dataPrepperConfiguration,
+            final Optional<Authenticator> optionalAuthenticator,
+            final HttpServer server
+    ) {
+        if (dataPrepperConfiguration.getMetricRegistryTypes().contains(MetricRegistryType.Prometheus)) {
+            final PrometheusMeterRegistry meterRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+            configureMetricRegistry(meterRegistry);
+
+            final PrometheusMetricsHandler metricsHandler = new PrometheusMetricsHandler(meterRegistry);
+
+            final List<HttpContext> contextList = new ArrayList<>(2);
```

Comment:
This bean creation has two distinct concerns involved: 1) Creating the `MeterRegistry` and 2) setting up the HttpServer to support it. It would be better to have a different construct handle the second concern. Perhaps there can be different beans for "server contexts" and then the `HttpServer` bean can assemble them all.